### PR TITLE
Support writes in mysql

### PR DESCRIPTION
**Pull request by mfouilleul**
Avoid trigging errors during insert, update, deletes.

### CODE REVIEW

File: redash/query_runner/mysql.py [hunk 1]
```diff
@@ -140,21 +140,39 @@ def run_query(self, query, user):
...
-            cursor.execute(query)
-
-            data = cursor.fetchall()
-
-            # TODO - very similar to pg.py
-            if cursor.description is not None:
-                columns = self.fetch_columns([(i[0], types_map.get(i[1], None)) for i in cursor.description])
-                rows = [dict(zip((c['name'] for c in columns), row)) for row in data]
-
+            queries = query.rstrip(';').split(';')
```
Comment (arikfr) [inline redash/query_runner/mysql.py]: That's not such a good idea, think of the following (contrived) example:

```
>>> "SELECT ';'".split(';')
["SELECT '", "'"]
```

But see the fix here: https://github.com/getredash/redash/pull/1955/commits/e02fdb3e37e5b6fefab591a44c74ecf4ce3b6223 for supporting multiple queries.

Once you rebase with latest, you can simplify the logic you implemented here.

File: redash/query_runner/mysql.py [hunk 2]
```diff
@@ -140,21 +140,39 @@ def run_query(self, query, user):
...
-            else:
-                json_data = None
+
+            if json_data is None:
                 error = "No data was returned."
+            else:
+                error = None
+
+            if transaction:
+                connection.commit()
```
Comment (arikfr) [inline redash/query_runner/mysql.py]: Why not commit above?

File: redash/query_runner/mysql.py [hunk 3]
```diff
@@ -140,21 +140,39 @@ def run_query(self, query, user):
...
+                if cursor.description is not None:
+                    columns = self.fetch_columns([(i[0], types_map.get(i[1], None)) for i in cursor.description])
+                    rows = [dict(zip((c['name'] for c in columns), row)) for row in data]
+                    data = {'columns': columns, 'rows': rows}
+                    json_data = json.dumps(data, cls=JSONEncoder)
+                    break
+                else:
+                    transaction = True
+                    columns = [{'name': 'Row(s) Affected',
+                                'type': types_map.get(str(type(rows_count)).upper(), None)}]
```
Comment (arikfr) [inline redash/query_runner/mysql.py]: Wouldn't it be always an integer? In which case you can use `TYPE_INTEGER` constant.

**Comment by mfouilleul** [comment]
Do we need more changes for this?

**Comment by arikfr** [comment]
Is this still needed following the changes in #3003?

**Comment by Jakdaw** [comment]
I'm not sure how much need there is for writes from redash - with the #3003 changes you can now achieve the same thing; eg with a redash query like:-

```
UPDATE mytable SET a=a+1 WHERE b={{whatever}};
SELECT ROW_COUNT() as "Row(s) Affected";
```

**Comment by arikfr** [comment]
Following #3003, closing this. @mfouilleul, if you still think this is needed (and willing to put in the work to update it to the changes in `redash.query_runner.mysql`) -- let me know. I will review it again before you do some extra work.

Thanks.

**Pull request closed without merging**